src/rest_api/amoc_api_logic_handler.erl

```diff
@@ -0,0 +1,84 @@
+%%==============================================================================
+%% Copyright 2015 Erlang Solutions Ltd.
+%% Licensed under the Apache License, Version 2.0 (see LICENSE file)
+%%==============================================================================
+-module(amoc_api_logic_handler).
+
+-behaviour(amoc_rest_logic_handler).
+
+-export([handle_request/3]).
+
+-spec handle_request(
+    OperationID :: amoc_rest_api:operation_id(),
+    Req :: cowboy_req:req(),
+    Context :: #{}
+) ->
+    {Status :: cowboy:http_status(), Headers :: cowboy:http_headers(), Body :: jsx:json_term()}.
+
+handle_request('NodesGet', _Req, _Context) ->
+    Status = amoc_cluster:get_status(),
+    Connected = maps:get(connected, Status, []),
+    FailedToConnect = maps:get(failed_to_connect, Status, []),
+    ConnectionLost = maps:get(connection_lost, Status, []),
+    Up = [{Node, <<"up">>} || Node <- [node() | Connected]],
+    DownNodes = lists:usort(FailedToConnect ++ ConnectionLost),
+    Down = [{Node, <<"down">>} || Node <- DownNodes],
+    ResponseList = Up ++ Down,
+    {200, #{}, [{<<"nodes">>, ResponseList}]};
+handle_request('ScenariosGet', _Req, _Context) ->
+    Scenarios = amoc_scenario:list_scenario_modules(),
+    BinaryScenarios = [atom_to_binary(S, latin1) || S <- Scenarios],
+    {200, #{}, [{<<"scenarios">>, BinaryScenarios}]};
+handle_request('StatusGet', _Req, _Context) ->
+    Apps = application:which_applications(),
+    Status = case lists:keyfind(amoc, 1, Apps) of
+                 {amoc, _Desc, _Vsn} -> <<"up">>;
+                 false -> <<"down">>
+             end,
+    {200, #{}, [{<<"node_status">>, Status}]};
+handle_request('ScenariosIdGet', _Req, #{id := Resource}) ->
+    Scenario = binary_to_atom(Resource, latin1),
+    case amoc_api_scenario_status:test_status(Scenario) of
+        doesnt_exist ->
+            {404, #{}, #{}};
+        Status ->
+            BinStatus = atom_to_binary(Status, latin1),
+            {200, #{}, [{<<"scenario_status">>, BinStatus}]}
+    end;
+handle_request('ScenariosIdPatch', _Req, #{'ScenarioExecution' := Body,
+                                           id := Resource }) ->
+    Scenario = binary_to_atom(Resource, latin1),
+    case amoc_api_scenario_status:test_status(Scenario) of
+        doesnt_exist ->
+            {404, #{}, #{}};
+        loaded ->
+            Users = maps:get(<<"users">>, Body),
+            SettingsMap = maps:get(<<"settings">>, Body, #{}),
+            Settings = [begin
+                            Key = binary_to_atom(K, latin1),
```

Comment:
This is probably a copy-paste. I think it'd be good to change `latin1` to `utf8` as we have in other projects (MongooseIM for instance). It'd be even better to use `binart_to_existing_atom`. The perfect solution would be to parse the binary and return atom only when the binary is acceptable.

Reply:
yes, I tried to introduce changes step by step. so the whole implementation of the rest api handlers is extracted from the previous cowboy-swagger based implementation. the risk is quite low, taking into account that this API should be called just once during the whole amoc app lifecycle. And the goal of this PR is to have equivalent API after migration, while binart_to_existing_atom usage requires introduction of some new error code. Also, `amoc_config_env:parse_value(V)` can still introduce new atoms, but we definitely don't want to change that functionality.